mermithid/processors/TritiumSpectrum/FakeDataGenerator.py

```diff
@@ -0,0 +1,253 @@
+        self.n_steps = reader.read_param(params, 'n_steps', 1000)
+        self.B_field = reader.read_param(params, 'B_field', 0.9578186017836624)
+
+#        #For Phase IV:
+#        #A_s = find_signal_activity(Nparticles, m, Q, Kmin) #Signal activity: events/s in ROI
+#        #S = A_s*runtime #Signal poisson rate
+#        self.S = reader.read_param(params, 'S', 2500)
+#        self.A_b = reader.read_param(params, 'A_b', 10**(-12)) #Flat background activity: events/s/eV
+#        self.B =self.A_b*self.runtime*(self.Kmax-self.Kmin) #Background poisson rate
+#        self.fb = self.B/(self.S+self.B) #Background fraction
+#        self.err_from_B = reader.read_param(params, 'err_from_B', 0.5) #In eV, kinetic energy error from f_c --> K conversion
+#
+#        #For a Phase IV gaussian smearing:
+#        self.sig_trans = reader.read_param(params, 'sig_trans', 0.020856) #Thermal translational Doppler broadening for atomic T (eV)
+#        self.other_sig = reader.read_param(params, 'other_sig', 0.05) #Kinetic energy broadening from other sources (eV)
+#        self.broadening = np.sqrt(self.sig_trans**2+self.other_sig**2) #Total energy broadening (eV)
+#
+#       #For binned spectrum generation (not implemented)
+#        self.bin_region_widths = reader.read_param(params, 'bin_region_width', [1., 9.]) #List of widths (eV) of regions where bin sizes do not change, from high to low energy
+#        self.nbins_per_region = reader.read_param(params, 'nbins_per_region', [300., 9.]) #List of number of bins in each of those regions, from high to low energy
+
+
+        # Phase II Spectrum parameters
+        self.runtime = reader.read_param(params, 'runtime', 31556952.) #In seconds
+        self.S = reader.read_param(params, 'S', 2500)
+        self.A_b = reader.read_param(params, 'A_b', 10**(-12)) #Flat background activity: events/s/eV
+        self.B =self.A_b*self.runtime*(self.Kmax-self.Kmin) #Background poisson rate
+        self.err_from_B = reader.read_param(params, 'err_from_B', 0.5) #In eV, kinetic energy error from f_c --> K conversion
+
+        #Simplified scattering model parameters
+        self.scattering_prob = reader.read_param(params, 'scattering_prob', 0.77)
+        self.scattering_sigma = reader.read_param(params, 'scattering_sigma', 18.6)
+        self.NScatters = reader.read_param(params, 'NScatters', 20)
+
+
+        #paths
+        self.simplified_scattering_path = reader.read_param(params, 'simplified_scattering_path', '/host/input_data/simplified_scattering_params.txt')
+        self.efficiency_path = reader.read_param(params, 'efficiency_path', '/host/input_data/combined_energy_corrected_eff_at_quad_trap_frequencies.json')
+
+
+        #options
+        self.use_lineshape = reader.read_param(params, 'use_lineshape', True)
+        self.detailed_or_simplified_lineshape = reader.read_param(params, 'detailed_or_simplified_lineshape', 'simplified')
```

Comment:
There should be a "value testing" for the key `detailed_or_simplified_lineshape` in case of a typo appearing here; otherwise if a typo or a misuse can lead to a silent error.

Reply:
added